financial-connections/src/main/java/com/stripe/android/financialconnections/features/common/LoadingContent.kt

```diff
@@ -137,7 +160,84 @@ internal fun LoadingSpinner(
     }
 }
 
-private const val LOADING_SPINNER_ROTATION_MS = 1000
+@Composable
+internal fun LoadingPill(
+    canShowPill: Boolean,
+    modifier: Modifier = Modifier,
+    showPillImmediately: Boolean = false,
+) {
+    val configuration = LocalConfiguration.current
+    val density = LocalDensity.current
+    val inspectionMode = LocalInspectionMode.current
+
+    val bottomOfScreen = remember(configuration, density) {
+        with(density) {
+            configuration.screenHeightDp.dp.roundToPx()
+        }
+    }
+
+    var showLoadingPill by rememberSaveable {
+        mutableStateOf(inspectionMode && showPillImmediately)
+    }
+
+    LaunchedEffect(canShowPill) {
+        if (canShowPill) {
+            delay(ShowLoadingPillDelayMillis)
+        } else {
+            delay(HideLoadingPillDelayMillis)
+        }
```

Comment:
We display the loading pill if loading hasn’t completed in 5 seconds. I also added a delay for hiding it, to make the transition smoother.